src/data/polynomial/partial_fractions.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2022 The Xena Project. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Sidharth Hariharan
+-/
+import ring_theory.localization.fraction_ring
+import data.polynomial.div
+import tactic.field_simp
+import tactic.linear_combination
+import data.zmod.basic
+import logic.function.basic
+
+/-!
+
+# Partial fractions
+
+These results were formalised by the Xena Project, at the suggestion
+of Patrick Massot.
+
+
+# The main theorem
+
+* `div_eq_quo_add_sum_rem_div`: General partial fraction decomposition theorem for polynomials over
+  an integral domain R :
+  If f, g₁, g₂, ..., gₙ ∈ R[X] and the gᵢs are all monic and pairwise coprime, then ∃ q, r₁, ..., rₙ
+  ∈ R[X] such that f / g₁g₂...gₙ = q + r₁/g₁ + ... + rₙ/gₙ and for all i, deg(rᵢ) < deg(gᵢ).
+
+* The result is formalized here in slightly more generality, using finsets. That is, if ι is an
+  arbitrary index type, g denotes a map from ι to R[X], and if s is an arbitrary finite subset of ι,
+  with g i monic for all i ∈ s and for all i,j ∈ s, i ≠ j → g i is coprime to g j, then we have
+  ∃ q ∈ R[X] , r : ι → R[X] such that ∀ i ∈ s, deg(r i) < deg(g i) and
+  f / ∏ g i = q + ∑ (r i) / (g i), where the product and sum are over s.
+
+* The proof is done by proving the two-denominator case and then performing finset induction for an
+  arbitrary (finite) number of denominators.
+
+## Scope for Expansion
+
+* Proving uniqueness of the decomposition
+
+-/
+
+
+variables (R : Type) [comm_ring R] [is_domain R]
+
+open_locale polynomial
+
+open polynomial
+
+variables (K : Type) [field K] [algebra R[X] K]  [is_fraction_ring R[X] K]
```

Comment:
```suggestion
variables (K : Type) [field K] [algebra R[X] K] [is_fraction_ring R[X] K]
```

Reply:
An alternative to working with an arbitrary `is_fraction_ring R[X] K` is to use `ratfunc` which is specifically intended to be the canonical field of rational functions. I expect it's not worth the effort to switch since it's easy enough to specialize to that field when needed.